lib/elixir/pages/Deprecations.md

```diff
@@ -0,0 +1,64 @@
+`Integer.to_char_list/1`  | [v1.5]        | `Integer.to_charlist/1` | v1.3
+`Kernel.to_char_list/1`   | [v1.5]        | `Kernel.to_charlist/1`  | v1.3
+`String.to_char_list/1`   | [v1.5]        | `String.to_charlist/1`  | v1.3
+`()` to mean `nil`        | [v1.5]        | `nil`                   | v1.0
+`:as_char_lists` value in `t:Inspect.Opts.t/0` type | [v1.5] | `:as_charlists` | v1.3
+`:char_lists` key in `t:Inspect.Opts.t/0` type | [v1.5] | `:charlists` | v1.3
+`char_list/0` type        | [v1.5]        | `charlist/0` type       | v1.3
+`@compile {:parse_transform, _}` in `Module` | [v1.5] | *None*      | *None*
+`Access.key/1`            | [v1.4]        | `Access.key/2`          | v1.3
+`Behaviour` module        | [v1.4]        | `@callback`             | v1.0
+`Enum.uniq/2`             | [v1.4]        | `Enum.uniq_by/2`        | v1.2
+`Float.to_char_list/2`    | [v1.4]        | `:erlang.float_to_list/2` | OTP 17
+`Float.to_string/2`       | [v1.4]        | `:erlang.float_to_binary/2` | OTP 17
+`HashDict` module         | [v1.4]        | `Map`                   | v1.2
+`HashSet` module          | [v1.4]        | `MapSet`                | v1.1
+`Set` module              | [v1.4]        | `MapSet`                | v1.1
+`Stream.uniq/2`           | [v1.4]        | `Stream.uniq_by/2`      | v1.2
+`IEx.Helpers.import_file/2` | [v1.4]  | [`IEx.Helpers.import_file_if_available/1`](https://hexdocs.pm/iex/IEx.Helpers.html#import_file_if_available/1) | v1.3
+`Mix.Utils.camelize/1`    | [v1.4]        | `Macro.camelize/1`      | v1.2
+`Mix.Utils.underscore/1`  | [v1.4]        | `Macro.underscore/1`    | v1.2
+Variable used as function call | [v1.4]   | Use parentheses         | v1.0
+Anoymous functions with no expression after `->` | [v1.4] | Use an expression or explicitely return `nil` | v1.0
+`Dict` module             | [v1.3]        | `Keyword`; `Map`        | v1.0 (`Keyword`); v1.2 (`Map`)
+`Keyword.size/1`          | [v1.3]        | `Kernel.length/1`       | v1.0
+`Map.size/1`              | [v1.3]        | `Kernel.map_size/1`     | v1.0
+`Set` behaviour           | [v1.3]        | `MapSet` data structure | v1.1
+`String.valid_character?/1` | [v1.3]      | `String.valid?/1`       | v1.0
+`Task.find/2`             | [v1.3]        | Use direct message matching | v1.0
+`:append_first` option in `Kernel.defdelegate/2` | [v1.3] | Define the function explicitely | *None*
+`/r` option in `Regex`    | [v1.3]        | `/U`                    | v1.1
+`\x{X*}` inside strings/sigils/charlists | [v1.3] | `\uXXXX`; `\u{X*}` | v1.1
+Map or dictionary as second argument in `Enum.group_by/3` | [v1.3] | Pass a function | v1.3
+Non-map as second argument in `URI.decode_query/2` | [v1.3] | Use a map | v1.0
+`Dict` behaviour          | [v1.2]        | *None*                  | *None*
```

Comment:
The replacement is the `Map` data structure (similar to Set being the MapSet data structure).